af/fastpath/fastpath/reprocessor.py

```diff
@@ -163,55 +199,69 @@ def score_measurement_and_upsert_fastpath(msm, msmt_uid, do_update: bool) -> Non
 @dataclass
 class Entity:
     jsonlf: Path
-    fd: int
+    fd: gzip.GzipFile
     jsonl_s3path: str
     lookup_list: list
 
 
 def finalize_jsonl(s3sig, db_conn, conf, e: Entity) -> None:
+    """For each JSONL file we do one upload to S3 and one
+    INSERT query with many rows
+    """
     jsize = int(e.fd.offset / 1024)
-    log.info(f"Closing jsonl and uploading it. Size: {jsize} KB")
+    log.info(f"Closing and uploading {e.jsonl_s3path}. Size: {jsize} KB")
     e.fd.close()
-    upload_to_s3(s3sig, conf.dst_bucket, e.jsonlf, e.jsonl_s3path)
-    update_db_table(db_conn, e.lookup_list, conf.jsonltbl)
+    stats["files_generated"] += 1
+    if conf.s3mode == "create":
+        upload_to_s3(s3sig, conf.dst_bucket, e.jsonlf, e.jsonl_s3path)
+    elif conf.s3mode == "check":
+        s3_check(s3sig, conf.dst_bucket, e.jsonlf, e.jsonl_s3path)
+    # update_db_table(db_conn, e.lookup_list, conf.jsonlmode)
+    update_jsonl_clickhouse_table(db_conn, e.lookup_list, conf.jsonlmode)
     e.jsonlf.unlink()
 
 
-def process_measurement(msm_tup, buf, seen_uids, conf, s3sig, db_conn):
+def process_measurement(can_fn, msm_tup, buf, seen_uids, conf, s3sig, db_conn):
     """Process a msmt
     If needed: create a new Entity tracking a jsonl file,
       close and upload jsonl to S3 and upsert db
     """
     THRESHOLD = 20 * 1024 * 1024
-    msm_jstr, msm, _ = msm_tup
+    msm_jstr, msm, msmt_uid = msm_tup
     if msm is None:
         msm = ujson.loads(msm_jstr)
-    if sorted(msm.keys()) == ["content", "format"]:
-        msm = unwrap_msmt(msm)
-
-    msmt_uid = trivial_id(msm)
-    rid = msm.get("report_id", None)
-    inp = msm.get("input", None)
-    tn = msm.get("test_name").replace("_", "")
-    cc = msm.get("probe_cc").upper()
-
-    # log.debug(f"Processing {msmt_uid} {tn} {cc} {rid} {inp}")
-    if msmt_uid in seen_uids:
-        log.info(f"DUPLICATE {msmt_uid} {tn} {cc} {rid} {inp}")
+    try:
+        if sorted(msm.keys()) == ["content", "format"]:
+            msm = unwrap_msmt(msm)
+
+        rid = msm.get("report_id")
+        test_name = msm.get("test_name")
+        input_, domain = extract_input_domain(msm, test_name)
+        assert not isinstance(input_, list)
+        tn = test_name.replace("_", "")
+        cc = msm.get("probe_cc").upper()
+        desc = f"{msmt_uid} {tn} {cc} {rid} {input_}"
+    except Exception as e:
+        log.info(f"Ignoring broken measurement")
```

Comment:
Can we add some additional logging information to identify which measurements are broken?

I think that once we do a run it would be useful to take a look at the broken ones and understand why we failed to load them.

I would say it's probably useful to record:
* Where it came from (the can or jsonl)
* The measurement_uid
* Increment a metric in statsd

Reply:
Added "{desc}"